packages/browser/src/plugins/remote-loader/index.ts

```diff
@@ -17,6 +28,120 @@ export interface RemotePlugin {
   settings: JSONObject
 }
 
+class ActionDestination implements Plugin {
+  name: string // destination name
+  version = '1.0.0'
+  type: Plugin['type']
+
+  alternativeNames: string[] = []
+
+  middleware: DestinationMiddlewareFunction[] = []
+
+  action: Plugin
+
+  constructor(name: string, action: Plugin) {
+    this.action = action
+    this.name = name
+    this.type = action.type
+    this.alternativeNames.push(action.name)
+  }
+
+  addMiddleware(...fn: DestinationMiddlewareFunction[]): void {
+    this.middleware.push(...fn)
+  }
+
+  private async transform(ctx: Context): Promise<Context> {
+    const modifiedEvent = await applyDestinationMiddleware(
+      this.name,
+      klona(ctx.event),
+      this.middleware
+    )
+
+    if (modifiedEvent === null) {
+      ctx.cancel(
+        new ContextCancelation({
+          retry: false,
+          reason: 'dropped by destination middleware',
+        })
+      )
+      return ctx
+    }
+
+    return new Context(modifiedEvent)
+  }
+
```

Comment:
Could we save some bytes by being more DRY?
Something like:
```ts
createMethod(methodName: 'track' | 'page' | 'identify' | 'alias' | 'group') {
    return async (ctx: Context): Promise<Context> => {
      if (!this.action[methodName]) return ctx

      const transformedContext = await this.transform(ctx)
      await this.action[methodName](transformedContext)

      return ctx
    }
  }
  
  alias = createMethod('track')
  group = createMethod('group')
   ... etc
```

Reply:
thanks!